DataFormats/EcalDigi/interface/EBDataFrame.h

```diff
@@ -3,36 +3,56 @@
 
 #include "DataFormats/EcalDetId/interface/EBDetId.h"
 #include "DataFormats/EcalDigi/interface/EcalDataFrame.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
 #include <iosfwd>
 
 
 
 /** \class EBDataFrame
       
 */
-class EBDataFrame : public EcalDataFrame 
-{
- public:
+class EBDataFrame : public EcalDataFrame {
+public:
   typedef EBDetId key_type; ///< For the sorted collection
   typedef EcalDataFrame Base;
 
-  EBDataFrame() {}
+  constexpr EBDataFrame() {}
   // EBDataFrame(DetId i) :  Base(i) {}
-  EBDataFrame(edm::DataFrame const & base) : Base(base) {}
-  EBDataFrame(EcalDataFrame const & base) : Base(base) {}
+  constexpr EBDataFrame(edm::DataFrame const & base) : Base(base) {}
+  constexpr EBDataFrame(EcalDataFrame const & base) : Base(base) {}
 
   /** estimator for a signal being a spike
    *  based on ratios between 4th, 5th and 6th sample
    */
-  float spikeEstimator() const;
+  constexpr float spikeEstimator() const
+  {
+        if ( size() != 10 ) {
+                edm::LogError("InvalidNumberOfSamples") << "This method only applies to signals sampled 10 times ("
```

Comment:
is this running on Device?

Reply:
nope. 

Initially just added constexpr to whatever was trivially possible to have constexpr. Then when during the compilation, noticed that EB/EE DataFrame do not need to be used (just EcalDataFrame is sufficient). was left in place.